builder/proxmox/iso/config.go

```diff
@@ -68,6 +78,90 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, []string, error) {
 		errs = packersdk.MultiErrorAppend(errs, errors.New("when specifying iso_url, iso_storage_pool must also be specified"))
 	}
 
+	// each device type has a maximum number of devices that can be attached.
+	// count iso, disks, additional isos configured for each device type, error if too many.
+	ideCount := 0
+	sataCount := 0
+	scsiCount := 0
+	virtIOCount := 0
+
+	if c.ISODevice == "" {
+		// set default ISO boot device ide2 if no value specified
+		log.Printf("iso_device not set, using default 'ide2'")
+		c.ISODevice = "ide2"
+		ideCount++
+		// Pass c.ISODevice value over to common config to ensure device index not used by disks
+		c.ISOBuilderCDROMDevice = "ide2"
+	} else {
```

Comment:
Should that be an `else`? If the above `if c.ISODevice == ""` branch executes, we can set the default values and use the same validation logic all the same I believe, this may be safe to let it be the normal execution path.